ultravox/training/helpers/prefetch_weights.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+from typing import List, Optional
+
+import huggingface_hub
+
+from ultravox.model import wandb_utils
+from ultravox.training import config_base
+
+ALLOW_PATTERNS = ["*.safetensors", "*.json"]
+
+
+def main(override_sys_args: Optional[List[str]] = None):
+    start = datetime.now()
+    print("Downloading weights ...")
+
+    args = config_base.get_train_args(override_sys_args)
+
+    for model_id in [args.text_model, args.audio_model]:
+        try:
+            huggingface_hub.snapshot_download(
```

Comment:
Also I forgot to mention: the reason for using `snapshot_download` over `AutoModel.from_pretrained` is that the download happens in parallel (~5 mins for the whole 70B).

I also tried [`hf-transfer`](https://github.com/huggingface/hf_transfer) that @petersalas mentioned, but even though each file was downloaded faster that way, no parallel download was happening so this one seemed to be faster overall in my tests.